### PR TITLE
[Core]Updating contact assembly

### PR DESCRIPTION
**Pull request by loumalouomega**
This PR is in two parts, one for the core and one for the ContactStructuralMechanicsApplication. Core Part

The assembling has been significantly reduced and now is done once per each pair.  A new condition is created for each pair and then destroyed.

### CODE REVIEW

**Comment by loumalouomega** [comment]
Complementary to #1196

File: kratos/processes/simple_mortar_mapper_process.cpp
Comment (RiccardoRossi) [inline kratos/processes/simple_mortar_mapper_process.cpp]: better use make_shared...

File: kratos/processes/simple_mortar_mapper_process.cpp
Comment (RiccardoRossi) [inline kratos/processes/simple_mortar_mapper_process.cpp]: call it InvertDiagonalMatrix, at least it is possible to understand what it does.

also i would make it to accept TMatrixType, as the only template parameter. You can get all the rest from that one.

**Review by RiccardoRossi — DISMISSED** [review]
i am approving, however please make another PR (or use this one) to fix my comments

**Comment by RiccardoRossi** [comment]
btw, are those functions tested? if so, please pass me the pointer to the tests. Otherwise: please add some tests (in the core, not in your application)

File: kratos/processes/simple_mortar_mapper_process.cpp
Comment (loumalouomega) [reply kratos/processes/simple_mortar_mapper_process.cpp]: OK, I was to avoid problem with the future migration of boost, but you are right

File: kratos/processes/simple_mortar_mapper_process.cpp
Comment (loumalouomega) [reply kratos/processes/simple_mortar_mapper_process.cpp]: Ok

**Comment by loumalouomega** [comment]
I will add some test that are right now in my application to the core

**Comment by loumalouomega** [comment]
Besides I will not merge until #1196 is approved too, or it will broke the contact application

**Comment by loumalouomega** [comment]
Additionally @RiccardoRossi 8b8a723b completely destroys my code, for me completely impossible to understand what I did in my own code, we need to thing about the styling before applying definetively

**Comment by loumalouomega** [comment]
Ping, remember #1196

**Comment by pooyan-dadvand** [comment]
I would definitely ask for some tests for the Kratos core parts. If they are in your application I would move them to the core to be with the code which is there.

**Comment by loumalouomega** [comment]
I added the corresponding cpp tests, the other tests are already in the core as python tests

**Comment by pooyan-dadvand** [comment]
Now I see it...

Another question, what is Ae? It would be better to give it a more descriptive name.

**Comment by loumalouomega** [comment]
It is a local matrix defined and used for the dual Lagrange multiplier
formulation. It is described in the mortar_classes.h file (where is
originally defined)

**Comment by pooyan-dadvand** [comment]
While looking for description of Ae (which is still not understandable for me....) I found several type_1, type_2 etc, and even worse size_1, size_2, ... variables which are not helping and even reducing the code readability. Giving better naming would improve the readability of your code and who knows, my help you find some size which is not used in correct way....

**Comment by loumalouomega** [comment]
Ok, I will use clearer names

**Comment by loumalouomega** [comment]
Done

**Review by pooyan-dadvand — DISMISSED** [review]
Thanks for considering the comments. Just to mention that I have not review the algorithms.

**Comment by loumalouomega** [comment]
Thanks I cannot merge until #1196 is approved too (or it will break the contact)

**Comment by loumalouomega** [comment]
@RiccardoRossi I updated some files and I need a new approval

**Review by RiccardoRossi — APPROVED** [review]
i would take the occasion to pass clang-format. If you tell me it is only formatting i'll accept it right away

**Comment by loumalouomega** [comment]
I will avoid the clang-format until @roigcarlo is satisfied with the result, certainly I am not happy with the results, last time I was completely unable to read my own code

**Pull request merged**
